src/main/java/sc/fiji/bdvpg/bdv/projector/AccumulateMixedProjectorARGB.java

```diff
@@ -127,26 +124,38 @@ else if ( projectionMode.equals( "Avg" ))
 
 	}
 
-	private String getProjectionMode( int sourceIndex, Map< SourceAndConverter, Map< String, Object > > sourceAndConverterToMetadata )
+	private String[] getProjectionModes( ArrayList< Source< ? > > sources )
 	{
-		Source< ? > source = sourceList.get( sourceIndex );
-		if ( source instanceof TransformedSource )
-			source = (( TransformedSource )source).getWrappedSource();
+		final List< SourceAndConverter > sacs = SourceAndConverterServices.getSourceAndConverterService().getSourceAndConverters();
+		final String[] projectionModes = new String[ sources.size() ];
 
-		String projectionMode = "Sum";
+		int sourceIndex = 0;
 
-		for ( SourceAndConverter sac : sourceAndConverterToMetadata.keySet() )
+		for ( Source<?> source : sources )
 		{
-			if ( sac.getSpimSource().equals( source ) )
-			{
-				final Set< String > metadata = sourceAndConverterToMetadata.get( sac ).keySet();
-				if ( metadata.contains( "ProjectionMode" ) )
-				{
-					projectionMode = (String) sourceAndConverterToMetadata.get( sac ).get( "ProjectionMode" );
-				}
-			}
+			if ( source instanceof TransformedSource )
```

Comment:
This, I don't like as well. What if you want different projectors for the Source and for the transformed Source ? Currently, no Source is wrapped as a TransformedSource in our design. So I would say it's not necessary. Have you tried to remove it? Does it change anything ?

Reply:
I agree, I also don't like it but I was not sure how this is going to be in the end with your code.
I can remove it!